RSSReader-Maruyama/Strategies/AppleAuthenticationStrategy.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  AppleAuthenticationStrategy.swift
+//  RSSReader-Maruyama
+//
+//  Created by 丸山翔 on 2020/10/02.
+//  Copyright © 2020 丸山翔. All rights reserved.
+//
+
+import UIKit
+import AuthenticationServices
+
+class AppleAuthenticationStrategy: NSObject, AuthenticationStrategy {
+    var delegate: AuthenticationDelegate
+    var presentingViewController: UIViewController
+    var handler: ((Result<User, Error>) -> Void)?
+    
+    init(delegate: AuthenticationDelegate, presentingViewController: UIViewController) {
+        self.presentingViewController = presentingViewController
+        self.delegate = delegate
+    }
+    
+    func login() {
+        let appleIDProvider = ASAuthorizationAppleIDProvider()
+        let request = appleIDProvider.createRequest()
+        request.requestedScopes = [.fullName, .email]
+        
+        let authorizationController = ASAuthorizationController(authorizationRequests: [request])
+        authorizationController.delegate = self
+        authorizationController.presentationContextProvider = self
+        authorizationController.performRequests()
+    }
+    
+    func logout(_ user: User) {
+    }
```

Comment:
実装を忘れているのか、あえて実装していないのかが分からないです。
あえて実装していないのであればコメントを残すようにしましょう。
```suggestion
    /// ToDo: 実装していない理由
    func logout(_ user: User) {
    }
```

Reply:
実装しました。